src/InputDataStream.cpp

```diff
@@ -39,6 +40,23 @@ void InputDataStream::loadTypeLib()
     m_type = m_registry->build(desc.getTypeName());
 }
 
+std::string InputDataStream::getMetadataEntry(const std::string& entry) const
+{
+    std::map<std::string, std::string>::const_iterator it = desc.getMetadataMap().find(entry);
+    if(it == desc.getMetadataMap().end())
+    {
+        std::string errorMessage = "Error: Logfile does not contain metadata " + entry + ". Maybe old logfile?";
+        std::cout << errorMessage << std::endl << "MetaData : " << std::endl;
+        for(auto e : desc.getMetadataMap())
+        {
+            std::cout << e.first << " : " << e.second << std::endl;
+        }
+        throw std::runtime_error("Error: Logfile does not contain metadata CXXType. Maybe old logfile?");
```

Comment:
1. why don't you reuse errorMessage here?
2. prefix Error does not hold much information if this is a runtime_error, better prefix with namespace, e.g., pocolog_cpp::InputDataStream::getMetadataEntry: